wordpress-mobile/WordPress-iOS
# Share Ext: Added notification support to the share extension

This PR adds notification support to the share extension. Because of the non-deterministic nature of _either_ WPiOS _or_ the extension getting the "did complete" callback, I needed to fire off notifs from 2 different places:
* `ExtensionNotificationManager.swift` (via share extension)
* `ShareExtensionSessionManager.swift` (via WPiOS)

![test2](https://user-images.githubusercontent.com/154014/35894652-7092b36c-0b79-11e8-83f6-67055ce5f665.gif)
![test3](https://user-images.githubusercontent.com/154014/35894656-75faf5a8-0b79-11e8-8f7e-9286fa859250.gif)

Two things to note here: 
1. The retry action is not implemented in this PR and will be part of the retry work itself.
2. IF the share extension's callback handles the post response...in-app notifs will not work. Perhaps @frosty could offer up some guidance here with his recent work on all this. :-)

Fixes #8146 

**Test 1:**
1. Open the Photos app and share 1 or 2 images.
2. Don't launch WPiOS.
3. You should see a iOS notification after a few seconds.
4. Tap the notification and it should launch WPiOS

**Test 2:**
1. Open the Photos app and share 1 or 2 images.
2. Quickly launch WPiOS after tapping "Publish" on the share extension.
3. You should see an in-app notification after a few seconds.
4. Tap the "Edit Post" action and verify it opens the post in question.

**Test 3:**
1. Open Safari and share a web page.
2. Don't launch WPiOS.
3. You should see a iOS notification after a few seconds.
4. Tap the notification and it should launch WPiOS.

@frosty Would you mind taking a peek at this? Thank you!!!!
